examples/hosts/app-integration/external-controller/src/app.ts

```diff
@@ -80,7 +82,12 @@ async function start(): Promise<void> {
 
     const div2 = document.createElement("div");
     contentDiv.appendChild(div2);
-    // We render a view which uses the controller.
+
+    // If this is a new container, we attach it to the client before rendering.
+    if (createNew) {clientResources.attach();}
```

Comment:
i moved it down purposefully to show that we can work with the container before it is actually attached, otherwise why not just automatically attach it.

Reply:
You're not setting any data between line 57 and 87. I would really love line 57 to be the following but I get if it doesn't work.

```typescript
clientResources = await client.createContainer({ id: containerId, logger: consoleLogger }, containerSchema).attach();
```

Reply:
Shouldn't this be awaited?

Reply:
i've tried with/without await, and it works the same each time. I wasn't sure either.

Reply:
oh yeah, this doesn't work because we're still awaiting for the return
![image](https://user-images.githubusercontent.com/1434956/124154997-e0453880-da4a-11eb-887a-880dfc68fce3.png)

but now that it has returned, we can call 'attach'

![image](https://user-images.githubusercontent.com/1434956/124155039-ec30fa80-da4a-11eb-80cf-4d7a20bc3698.png)